### ISSUE TITLE
`Cue eval` hangs on cyclic package imports

### ISSUE THREAD

**Issue by movatechtoday** [cue-lang/cue#2652]
### What version of CUE are you using (`cue version`)?

<pre>
$ cue version

cue version v0.6.0

go version go1.20.3
      -buildmode exe
       -compiler gc
     CGO_ENABLED 1
          GOARCH amd64
            GOOS linux
         GOAMD64 v1

</pre>

### Does this issue reproduce with the latest stable release?

Yes

### What did you do?
I create a module with the following structure.

<pre>
.
├── cue.mod
│   └── module.cue
└── main
    ├── libs.cue
    └── main.cue
</pre>

**module.cue**
<pre>
module: "cuefy.today/tests"
</pre>

**main/libs.cue**
<pre>
package libs

field: "field"
</pre>

**main/main.cue**

<pre>
package main

import (
    "cuefy.today/tests/main:libs"
    "cuefy.today/tests/main"
)

field_001: libs.field   // <-- import from the libs package
field_main: "field_main"
field_main_ref: main.field_main // <-- import from the main package
</pre>

I mistakenly created a cycle inside the `main` package to the `main` package. 

And run the following command: 
```shell
 cue eval main/main.cue 
```
 

### What did you expect to see?

After running the following commad:
```shell
 cue eval main/main.cue 
```

I would like to see an error message that a cycle detected. 

### What did you see instead?

`cue eval` hangs

**Comment by mvdan** [comment]
Have you tried master? This sounds like a variation of #849, which was fixed shortly after our last stable release.

**Comment by movatechtoday** [comment]
No, I tried the latest stable version. Let me check with on the master branch

**Comment by mvdan** [comment]
Per the above, I'm going to assume this was indeed fixed. Let me know if not.
